src/System.Collections/tests/Performance/Perf.Dictionary.cs

```diff
@@ -0,0 +1,195 @@
+                dict.Add(key, PerfUtils.CreateString(50));
+            }
+            return dict;
+        }
+
+        [Benchmark]
+        [MemberData("TestDictionaries")]
+        public void Add(Dictionary<string, string> dict)
+        {
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    dict.Add("key1", "string"); dict.Add("key2", "string"); dict.Add("key3", "string");
+                    dict.Add("key4", "string"); dict.Add("key5", "string"); dict.Add("key6", "string");
+                    dict.Add("key7", "string"); dict.Add("key8", "string"); dict.Add("key9", "string");
+                }
+                for (int i = 1; i <= 9; i++)
+                    Assert.True(dict.Remove("key" + i));
+            }
+        }
+
+        [Benchmark]
+        public void ctor()
+        {
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                {
+                    new Dictionary<int, string>(); new Dictionary<int, string>(); new Dictionary<int, string>();
+                    new Dictionary<int, string>(); new Dictionary<int, string>(); new Dictionary<int, string>();
+                    new Dictionary<int, string>(); new Dictionary<int, string>(); new Dictionary<int, string>();
+                }
+        }
+
+        [Benchmark]
+        [InlineData(0)]
+        [InlineData(1024)]
+        [InlineData(4096)]
+        [InlineData(16384)]
+        public void ctor_int(int size)
+        {
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                {
+                    new Dictionary<int, string>(size); new Dictionary<int, string>(size); new Dictionary<int, string>(size);
+                    new Dictionary<int, string>(size); new Dictionary<int, string>(size); new Dictionary<int, string>(size);
+                    new Dictionary<int, string>(size); new Dictionary<int, string>(size); new Dictionary<int, string>(size);
+                }
+        }
+
+        [Benchmark]
+        [MemberData("TestDictionaries")]
+        public void GetItem(Dictionary<string, string> dict)
+        {
+            // Setup
+            string retrieved;
+            for (int i = 1; i <= 9; i++)
+                dict.Add("key" + i, "value");
+
+            // Actual perf testing
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    retrieved = dict["key1"]; retrieved = dict["key2"]; retrieved = dict["key3"];
+                    retrieved = dict["key4"]; retrieved = dict["key5"]; retrieved = dict["key6"];
+                    retrieved = dict["key7"]; retrieved = dict["key8"]; retrieved = dict["key9"];
+                }
+            }
+
+            // Teardown
```

Comment:
Is this teardown necessary?  It looks like we're creating new dictionary instances for every invocation of the test.